AirCasting/SessionViews/SessionHeaderView.swift

```diff
@@ -72,24 +72,33 @@ private extension SessionHeaderView {
             //  |_________|     |-------------------|
             // so the idea at leat for now is this below
             #warning("Fix - Handle session.deviceType (for now it is always nill)")
-            switch session.type?.description {
-            case "Fixed":
-                Text("\(session.type!.description) : AirBeam3")
-                    .font(Font.moderate(size: 13, weight: .regular))
-            case "Mobile":
-                if session.allStreams!.count > 1 {
-                    Text("\(session.type!.description): AirBeam3")
-                        .font(Font.moderate(size: 13, weight: .regular))
+            sensorType
+                .font(Font.moderate(size: 13, weight: .regular))
+        }
+        .foregroundColor(.darkBlue)
+    }
+    
+    var sensorType: some View {
+        var sensorNames = [String]()
+        var text = ""
+        for session in session.allStreams! {
+            if var name = session.sensorPackageName {
+                if name.contains(":") {
+                    name = name.components(separatedBy: ":").first!
                 } else {
-                    Text("\(session.type!.description): Phone Mic")
-                        .font(Font.moderate(size: 13, weight: .regular))
+                    name = name.components(separatedBy: "-").first!
                 }
-            default:
-                Text(session.deviceType?.description ?? "")
-                    .font(Font.moderate(size: 13, weight: .regular))
+                (name == "Builtin") ? (name = "Phone mic") : (name = name)
+                !sensorNames.contains(name) ? sensorNames.append(name) : nil
             }
         }
-        .foregroundColor(.darkBlue)
+        for sensor in sensorNames {
```

Comment:
no need for this loop, just use `sensorNames.joined(separator: ", ")`

Reply:
WOW, Cool!